sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -213,30 +206,30 @@ class ProjectOverviewView extends VerticalLayout{
                 .setCaption("Project Code").setId("ProjectCode").setMaximumWidth(
                 MAX_CODE_COLUMN_WIDTH)
         projectGrid.addColumn({ it.title })
-                .setCaption("Project Title").setId("ProjectTitle").setDescriptionGenerator({it.title})
+                .setCaption("Project Title").setId("ProjectTitle").setDescriptionGenerator({ProjectSummary project -> project.title})
 
         ValueProvider<ProjectSummary, RelativeCount> receivedProvider = { ProjectSummary it ->
             new RelativeCount(it.samplesReceived.passingSamples, it.totalSampleCount )
         }
-        projectGrid.addColumn(receivedProvider).setStyleGenerator({getStyleForColumn(it, receivedProvider)})
+        projectGrid.addColumn(receivedProvider).setStyleGenerator({ProjectSummary project -> getStyleForColumn(project, project.samplesReceived)})
                 .setCaption("Samples Received").setId("SamplesReceived")
 
         ValueProvider<ProjectSummary, RelativeCount> failedQcProvider = { ProjectSummary it ->
             new RelativeCount(it.samplesQc.passingSamples, it.totalSampleCount )
         }
-        projectGrid.addColumn(failedQcProvider)
-                .setCaption("Samples Passed QC").setId("SamplesPassedQc").setStyleGenerator({getStyleForFailureColumn(it, failedQcProvider)})
+        projectGrid.addColumn(failedQcProvider).setStyleGenerator({ProjectSummary project -> getStyleForColumn(project, project.samplesQc)})
+                .setCaption("Samples Passed QC").setId("SamplesPassedQc")
 
         ValueProvider<ProjectSummary, RelativeCount> libraryPrepProvider = {ProjectSummary it ->
             new RelativeCount(it.samplesLibraryPrepFinished.passingSamples , it.totalSampleCount)
         }
-        projectGrid.addColumn(libraryPrepProvider)
-                .setCaption("Library Prep Finished").setId("LibraryPrepFinished").setStyleGenerator({getStyleForColumn(it, libraryPrepProvider)})
+        projectGrid.addColumn(libraryPrepProvider).setStyleGenerator({ProjectSummary project -> getStyleForColumn(project, project.samplesLibraryPrepFinished)})
+                .setCaption("Library Prep Finished").setId("LibraryPrepFinished")
 
         ValueProvider<ProjectSummary, RelativeCount> dataAvailableProvider = { ProjectSummary it ->
             new RelativeCount(it.sampleDataAvailable.passingSamples , it.totalSampleCount)
         }
-        projectGrid.addColumn(dataAvailableProvider).setStyleGenerator({getStyleForColumn(it, dataAvailableProvider)})
+        projectGrid.addColumn(dataAvailableProvider).setStyleGenerator({ProjectSummary project -> getStyleForColumn(project, project.sampleDataAvailable)})
```

Comment:
The reason the data provider of the column was used here before was that we wanted to make sure that colouring and value of the column do not go out of sync. I think this solution works as well but you have to be extra careful that you provide the correct information here. This could potentially become a problem if you wanted to decide the colouring based on the provided value rather than the row object. For now, this increases readability, I like it.